source/module_hsolver/hsolver_pw.cpp

```diff
@@ -574,7 +617,7 @@ void HSolverPW<T, Device>::update_precondition(std::vector<Real>& h_diag, const
 
             if (this->method == "dav_subspace")
             {
-                h_diag[ig] = g2kin;
+                h_diag[ig] = g2kin + Real(elecstate::PotLocal::get_vl_of_0());
```

Comment:
would it be better to pass in the vl(0) value from the parameters of this function?